packages/react-native-jigsaw/src/core/DataContext.tsx

```diff
@@ -0,0 +1,20 @@
+import * as React from "react";
+
+export type DataContextType = {
+  signInWithEmailAndPassword: (
+    email: string,
+    password: string
+  ) => Promise<void>;
+  signOut: () => Promise<void>;
+};
+
+export const defaultDataContext: DataContextType = {
+  signInWithEmailAndPassword: () => {
+    throw new Error("Signin has not been implemented for this app");
+  },
+  signOut: () => {
+    throw new Error("Signout has not been implemented for this app");
+  },
+};
+
+export const DataContext = React.createContext(defaultDataContext);
```

Comment:
I think something like this is a really good idea, I only want to ask how tightly you think your changes might be tied to this exact API, because this is something I think we should align on as an engineering team.

If it's isolated to firebase stuff and you're pretty confident it can be unwound and are just experiment then :rocket:

Reply:
Totally, this is all subject to change going forward. For the time being it's pretty much just exploratory